tests/playwright/src/model/pages/resource-card-page.ts

```diff
@@ -0,0 +1,40 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { expect as playExpect, type Locator, type Page } from '@playwright/test';
+
+import { ResourcesPage } from './resources-page';
+
+export abstract class ResourceCardPage extends ResourcesPage {
+  readonly resourceCard: Locator;
+  readonly providerSetup: Locator;
+  readonly providerConnections: Locator;
+
+  constructor(page: Page, resourceName: string) {
+    super(page);
+    this.resourceCard = this.featuredProviderResources.getByRole('region', { name: resourceName, exact: true });
+    this.providerSetup = this.resourceCard.getByRole('region', { name: 'Provider Setup', exact: true });
+    this.providerConnections = this.resourceCard.getByRole('region', { name: 'Provider Connections', exact: true });
+  }
+
+  public async clickOnProviderSetupButton(buttonLabel: string): Promise<void> {
```

Comment:
I kind of dislike this implementation because it's not obvious that you need to provide the aria-label to the method in order for it to work, this needs to be made clearer.